pkg/cloudprovider/controller/nodecontroller.go

```diff
@@ -560,7 +595,7 @@ func (nc *NodeController) MonitorNodeStatus() error {
 				nc.deletePods(node.Name)
 			}
 			if lastReadyCondition.Status == api.ConditionUnknown &&
-				nc.now().After(lastReadyCondition.LastProbeTime.Add(nc.podEvictionTimeout-gracePeriod)) {
+				nc.now().After(nc.nodeStatusMap[node.Name].timestamp.Add(nc.podEvictionTimeout-gracePeriod)) {
```

Comment:
Can't we just maintain a separate map for LastTransitionTime, and update it with local now whenever we receive a change in LastTransitionTime from Kubelet (similar to how you are already handling LastProbeTime)?